### DISCUSSION TITLE
Is it possible to build a portable executable?

### DISCUSSION THREAD

**Discussion by Kasea** [microsoft/react-native-windows#9332] [Q&A] [answered]
Like the title says, is it possible to build a portable executable using RNW?

I am aware you can make a standalone one where you create a APPX package, but I would much rather prefer a portable executable. Is this possible?

**Comment by chrisglein** [comment, marked as answer]
RNW apps are UWP apps by default. There are some considerations to that for sure (https://microsoft.github.io/react-native-windows/docs/native-code#win32-desktop-apps-vs-rnw-apps), including the fact that UWP apps aren't single EXEs and have a totally different packaging model.

It's possible to add RNW to a classic Win32 app in an "island". You can find some info on that here: https://github.com/microsoft/react-native-windows-samples/issues/540. There may be additional considerations if you're looking to make a single file EXE with everything bundled together, not sure about that.

**Reply by AlbertShown** [reply]
> if you're looking to make a single file EXE with everything bundled together, not sure about that.

Using [WebUI](https://github.com/webui-dev/webui/tree/main/examples/C/react), you can make a full portable single file EXE, around 1 MB.

**Comment by raffaeler** [comment]
@chrisglein This is a prescriptive requirement for many large customers of mine.
They banned internally the MS Store and do not allow UWP/Windows Apps to be installed on their clients.
Currently they are deploying with either msi packages or Clickonce.

Is there any hope to see classic deployment? This is a huge restriction that won't ever be accepted (they are more willing to convert to a web application rather than changing this decision).

**Comment by nocanstillbb** [comment]
now react native for windows can use win app sdk not uwp  in new arch ,  Is it possible to build a portable executable?

**Reply by chrisglein** [reply]
Ah yes, much has changed over the past few years since this discussion post was created. Yes, on new architecture RNW is now a Win32+WinAppSDK app, not UWP. So that means this needs fresh evaluation. I'm not personally familiar with whether one can create a single bundled executable of a Win32+WinAppSDK app, but that would be the deciding question. I suspect the answer is yes but I'm not familiar with the steps.

**Reply by nocanstillbb** [reply]
https://microsoft.github.io/react-native-windows/docs/new-architecture
```bash
https://microsoft.github.io/react-native-windows/docs/new-architecture
```
still create a uwp app which can't  publish to portable dir

**Reply by chrisglein** [reply]
If you follow the steps to [create a new arch application](https://microsoft.github.io/react-native-windows/docs/new-architecture#creating-a-new-architecture-application) that will be a Win32 app with WinAppSDK, not a UWP app.

**Reply by nocanstillbb** [reply]
> If you follow the steps to [create a new arch application](https://microsoft.github.io/react-native-windows/docs/new-architecture#creating-a-new-architecture-application) that will be a Win32 app with WinAppSDK, not a UWP app.


<paaname>.package is a uwp shell that using  winappsdk nuget package , 

I tried to implement the unpackaged winsdkapp based on this [unpacked demo](https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/Unpackaged/cpp-console-unpackaged) and found that the project structure was different

**Reply by nocanstillbb** [reply]
However, I am not familiar with UWP and Win32 App SDK projects. I will study it again. Thanks.
